### PR TITLE
Fix19817

### LINKED ISSUE #19817: LoadWithPartialName should not throw according to documentation
_From @LasseNisted on August 28, 2018 11:35_

According to the documentation at
https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.loadwithpartialname?view=netcore-2.1
Assembly.LoadWithPartialName should not throw a FileNotFoundException. And this is also the case in .NET Framework 4.7.1. However, in .NET Core 2.1 it is implemented via Assembly.Load which throws. The documentation or implementation should be updated.

_Copied from original issue: dotnet/corefx#31999_

### PR DESCRIPTION
**Pull request by sdmaclea**
Assembly.LoadFromPartialName(string) should not throw FileNotFoundException
rather it should return null.

Fixes #19817

### CODE REVIEW

**Comment by sdmaclea** [comment]
I plan to add xunit tests to corefx

**Comment by sdmaclea** [comment]
test Ubuntu x64 Formatting
test Windows_NT x64 Release CoreFX Tests
test Ubuntu x64 Checked CoreFX Tests

File: src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs [hunk 1]
```diff
@@ -205,7 +205,17 @@ public static Assembly LoadWithPartialName(string partialName)
             if (partialName == null)
                 throw new ArgumentNullException(nameof(partialName));
 
-            return Load(partialName);
+            if ((partialName.Length == 0) || (partialName[0] == '\0'))
```
Comment (vitek-karas) [inline src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs]: `string.IsNullOrEmpty`

Comment (vitek-karas) [reply src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs]: Also - I think the Length will be zero if the first character is the nul terminator.

**Pull request merged**